test/test_eos_tabulated.cpp

```diff
@@ -257,6 +262,98 @@ SCENARIO("SpinerEOS depends on rho and sie", "[SpinerEOS],[DependsRhoSie]") {
+      REQUIRE(eospac_size == eospac_orig.SerializedSizeInBytes());
+
+      const std::size_t rhoT_shared_size = rhoT_orig.DynamicMemorySizeInBytes();
+      REQUIRE(rhoT_size > rhoT_shared_size);
+
+      const std::size_t rhoSie_shared_size = rhoSie_orig.DynamicMemorySizeInBytes();
+      REQUIRE(rhoSie_size > rhoSie_shared_size);
+
+      const std::size_t eospac_shared_size = eospac_orig.DynamicMemorySizeInBytes();
+      REQUIRE(eospac_size > eospac_shared_size);
+
+      THEN("We can deserialize into shared memory") {
+        using singularity::SharedMemSettings;
+        using RhoTTricks = singularity::table_utils::SpinerTricks<SpinerEOSDependsRhoT>;
+        using RhoSieTricks =
+            singularity::table_utils::SpinerTricks<SpinerEOSDependsRhoSie>;
+
+        char *rhoT_shared_data = (char *)malloc(rhoT_shared_size);
+        char *rhoSie_shared_data = (char *)malloc(rhoSie_shared_size);
+        char *eospac_shared_data = (char *)malloc(eospac_shared_size);
+
+        SpinerEOSDependsRhoT eos_rhoT;
+        std::size_t read_size_rhoT =
+            eos_rhoT.DeSerialize(rhoT_data, SharedMemSettings(rhoT_shared_data, true));
+        REQUIRE(read_size_rhoT == rhoT_size);
+        REQUIRE(RhoTTricks::DataBoxesPointToDifferentMemory(rhoT_orig, eos_rhoT));
+
+        SpinerEOSDependsRhoSie eos_rhoSie;
+        std::size_t read_size_rhoSie = eos_rhoSie.DeSerialize(
+            rhoSie_data, SharedMemSettings(rhoSie_shared_data, true));
+        REQUIRE(read_size_rhoSie == rhoSie_size);
+        REQUIRE(RhoSieTricks::DataBoxesPointToDifferentMemory(rhoSie_orig, eos_rhoSie));
+
+        eospac_orig.Finalize();
+        EOS eos_eospac = EOSPAC();
+        std::size_t read_size_eospac = eos_eospac.DeSerialize(
+            eospac_data, SharedMemSettings(eospac_shared_data, true));
+        REQUIRE(read_size_eospac == eospac_size);
+
+        AND_THEN("EOS lookups work") {
+          constexpr Real rho_trial = 1;
+          constexpr Real sie_trial = 1e12;
+          const Real P_eospac =
+              eos_eospac.PressureFromDensityInternalEnergy(rho_trial, sie_trial);
+          const Real P_spiner_orig =
+              rhoT_orig.PressureFromDensityInternalEnergy(rho_trial, sie_trial);
+          const Real P_spiner_rhoT =
+              eos_rhoT.PressureFromDensityInternalEnergy(rho_trial, sie_trial);
+          const Real P_spiner_rhoSie =
+              eos_rhoSie.PressureFromDensityInternalEnergy(rho_trial, sie_trial);
+          REQUIRE(isClose(P_eospac, P_spiner_orig));
+          REQUIRE(isClose(P_eospac, P_spiner_rhoT));
+          REQUIRE(isClose(P_eospac, P_spiner_rhoSie));
+        }
+
+        eos_eospac.Finalize();
+        free(rhoT_shared_data);
+        free(rhoSie_shared_data);
+        free(eospac_shared_data);
+      }
+      free(rhoT_data);
+      free(rhoSie_data);
+      free(eospac_data);
+    }
+
+    rhoT_orig.Finalize();
+    rhoSie_orig.Finalize();
```

Comment:
Maybe I knew this once and forgot, but I'm realizing that this will likely require substantial modifications in the fortran interface to account for the new ways of initialization and finalization.

Reply:
We will need to expose a serialization/deserialization interface on the fortran side, yes. That will need to be on the tasking for next FY.

Reply:
Good point! I'll start thinking about this when the work packages are more finalized.